core/src/main/scala/org/apache/spark/shuffle/ExternalShuffleService.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import org.apache.spark.SparkConf
+import org.apache.spark.SecurityManager
+import org.apache.spark.network.netty.SparkTransportConf
+import org.apache.spark.network.shuffle.ExternalShuffleBlockHandler
+import org.apache.spark.network.TransportContext
+import org.apache.spark.network.sasl.SaslRpcHandler
+import org.apache.spark.network.server.TransportServer
+import org.apache.spark.Logging
+import org.apache.spark.util.Utils
+import sun.misc.Signal
+import sun.misc.SignalHandler
+import java.util.concurrent.CountDownLatch
+
+/**
+ * A main class for runing the external shuffle service. This follows
+ * closely how [[org.apache.spark.deploy.worker.StandaloneWorkerShuffleService]] works.
+ *
+ * @note This adds handlers for SIGINT and SIGTERM to terminate gracefully.
+ */
+object ExternalShuffleService extends Logging {
```

Comment:
How about a compromise, of renaming `StandaloneWorkerShuffleService` to `ExternalShuffleService` (since it has no standalone-specific references anyway) and adding a main method there? See my comment below.